docusaurus-docs/conda-store/references/faq.md

```diff
@@ -94,19 +94,50 @@ It consists of:
 3. the id of a build
 4. the environment name.
 
-The version 2 format is now the default. Environments created using the version
-1 format will continue to be accessible in the UI, but new builds will use the
-version 2 format. No changes are needed for existing deployments of conda-store.
+However, version 2 build paths don't solve the problem completely because they
+include user-provided data, like the environment name, and that data can be
+arbitrary large.
 
-There is no real reason to use the version 1 format anymore, but it can be
+To solve this problem, version 3 was introduced, which will always have the same
+size. It looks like this:
+
+```bash
+64a943764b70e8fe181643404894f7ae
+```
+
+It's a truncated SHA-256 hex digest, which is calculated based on:
+
+- namespace name
+- specification hash (also SHA-256)
+- build timestamp
+- build id.
+
+See `BuildKey._version3_fmt` for details.
+
+:::note
+When version 3 is used, `Build.build_path` will not include the namespace name,
+because it's not fixed size, so all builds will be placed right into
+`CondaStore.store_directory`. Additionally, `CondaStore.environment_directory`
+will be completely ignored, so no symlinks will be created, because the
+environment directory format also includes variable-size data (the namespace and
+environment names). This shouldn't impact anything because the generated
+artifacts rely on storage or use the database, and those are not affected by
+these changes to the state directory.
+:::
```

Comment:
> so no symlinks will be created [...] This shouldn't impact anything [...]

So, I'm actually concerned about this. And I think it *will* impact Nebari/jhub integration because I see the following in `examples/docker/assets/jupyterhub_config.py`, but I have no way to test this right now:

```py
with open(os.path.join(home, '.condarc'), 'w') as f:
    f.write(json.dumps({
        'envs_dirs': [
            '/opt/conda-store/conda-store/default/envs',
            '/opt/conda-store/conda-store/filesystem/envs',
            f'/opt/conda-store/conda-store/{self.user.name}/envs',
        ]
    }))
```

The question is how to proceed. As explained in the doc note above, `environment_directory` is ignored because there's variable-size data, which is incompatible with the purpose of v3. So I see these options:
- Keep v2 the default because it's a good balance (no big hashes and supports env symlinking), v3 will be there for those who don't need to rely on envs_dirs
- Set v2 only in the docker config files bundled with conda-store, which is what Nebari uses.

Reply:
@jaimergp Let's do the final review later. First, I want to agree on the plan here. You can ignore everything else.

Reply:
Update: I've scheduled a meeting with Chuck to discuss how Nebari uses this.

Reply:
Tag me again after the meeting and I'll review.